src/view/widgets/Tab.ts

```diff
@@ -0,0 +1,270 @@
+import { Option } from "../../utils/monads";
+import { Resources } from "../Package";
+import { GetWidgetProperties, MergeProperties, Widget } from "../Widget";
+import { Badge, Element } from "./Basic";
+import { IconButton } from "./Button";
+import { EventHandler, EventWidget } from "./Interactive";
+import { CustomTextElement } from "./Text";
+
+export class Tab extends CustomTextElement {
+
+    public static readonly RESOURCES = Resources.fromResources({
+        // For some reason MS fast just can't figure out how to move that damn underline so I'll just unMS it
+        css: `
```

Comment:
Basically, if you start adding tabs later on without just completely removing and adding back the tabs, MS fast gets confused about the locations of the tabs and starts showing UB as it sometimes works and sometimes just moves to a semi random spot.